backend/test/modules/tournament-tree/core/domain/TournamentTree.spec.ts

```diff
@@ -1,118 +1,349 @@
+      //Given
+      const teamEntityIdGen = NumberIdGeneratorStub(1000, 'team');
+      const matchEntityIdGen = NumberIdGeneratorStub(1000, 'match');
+      const tournamentTeamsList = generateTournamentTeamsList(teamEntityIdGen, 6);
+
+      // //When
+      const tournamentTree: TournamentTree = TournamentTree.createSingleTournamentTree({
+        tournamentId: testTournamentId.generate(),
+        tournamentTeams: tournamentTeamsList,
+        entityIdGenerator: matchEntityIdGen,
+      });
+
+      const expectedWinnerTree: FightingTeamsGroup[] = [
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_7'),
+          firstTeam: tournamentTeamsList[0],
+          secondTeam: undefined,
+          fightingTeamsGroupLevel: 0,
+          nextMatchId: FightingTeamsGroupId.from('match_3'),
+          matchNumberInSequence: 1,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_6'),
+          firstTeam: tournamentTeamsList[4],
+          secondTeam: tournamentTeamsList[3],
+          fightingTeamsGroupLevel: 0,
+          nextMatchId: FightingTeamsGroupId.from('match_3'),
+          matchNumberInSequence: 2,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_5'),
+          firstTeam: tournamentTeamsList[2],
+          secondTeam: tournamentTeamsList[5],
+          fightingTeamsGroupLevel: 0,
+          nextMatchId: FightingTeamsGroupId.from('match_2'),
+          matchNumberInSequence: 3,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_4'),
+          firstTeam: undefined,
+          secondTeam: tournamentTeamsList[1],
+          fightingTeamsGroupLevel: 0,
+          nextMatchId: FightingTeamsGroupId.from('match_2'),
+          matchNumberInSequence: 4,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_3'),
+          firstTeam: undefined,
+          secondTeam: undefined,
+          fightingTeamsGroupLevel: 1,
+          nextMatchId: FightingTeamsGroupId.from('match_1'),
+          matchNumberInSequence: 5,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_2'),
+          firstTeam: undefined,
+          secondTeam: undefined,
+          fightingTeamsGroupLevel: 1,
+          nextMatchId: FightingTeamsGroupId.from('match_1'),
+          matchNumberInSequence: 6,
+        },
+        {
+          fightingTeamsGroupId: FightingTeamsGroupId.from('match_1'),
+          firstTeam: undefined,
+          secondTeam: undefined,
+          fightingTeamsGroupLevel: 2,
+          nextMatchId: undefined,
+          matchNumberInSequence: 7,
+        },
+      ].map((item) => FightingTeamsGroup.fromObj(item));
+
+      // Then
+      expect(tournamentTree.getTournamentTreeArray()).toIncludeSameMembers(expectedWinnerTree);
+    });
+  });
+
+  describe('getMatchQueue() function', () => {
```

Comment:
Nazwy funkcji, itp moga sie zmieniac a funkcjonalnosci pozostawac takie same. Wiec dobrze opisywac testy biznesowo, a nie "technicznie". Moga byc w formie given-when-then nazwy, should albo tytułów scenariuszy.

Wiec np w desciribe moze byc:
`Calculating matches queue`
a w it `from 4 teams tree`

Reply:
Masz rację, nie pomyślałem o tym. naprawię to